Xamarin.Forms.Platform.Android/CollectionView/ItemContentView.cs

```diff
@@ -0,0 +1,57 @@
+using Android.Content;
+using Android.Views;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	internal class ItemContentView : ViewGroup
+	{
+		protected readonly IVisualElementRenderer Content;
+
+		public ItemContentView(IVisualElementRenderer content, Context context) : base(context)
+		{
+			Content = content;
+			AddContent();
+		}
+
+		void AddContent()
+		{
+			AddView(Content.View);
+		}
+
+		protected override void OnLayout(bool changed, int l, int t, int r, int b)
+		{
+			var size = Context.FromPixels(r - l, b - t);
+
+			Content.Element.Layout(new Rectangle(Point.Zero, size));
+
+			Content.UpdateLayout();
+		}
+
+		protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
+		{
+			int pixelWidth = MeasureSpec.GetSize(widthMeasureSpec);
+			int pixelHeight = MeasureSpec.GetSize(heightMeasureSpec);
+
+			var width = Context.FromPixels(pixelWidth);
+			var height = Context.FromPixels(pixelHeight);
+
+			SizeRequest measure = Content.Element.Measure(width, height, MeasureFlags.IncludeMargins);
+
+			if (pixelWidth == 0)
```

Comment:
I haven't done much measuring/layout in forms, but why do we only update if == 0? Won't this always use the old sizes?

Reply:
The layout manager will give us a size of 0 in an unconstrained direction. E.g., a vertically scrolling LinearLayoutManager will always give us a height of 0 (because it doesn't care; the item can be any height and the layout simply grows in that direction). So in that direction, we go ahead and measure the desired height of our Forms view. If it gives us a non-zero value, that's the maximum allotted space in that direction, and we will lay the Forms control out in that space regardless of size.